### DISCUSSION TITLE
eth_sign method fail with @walletconnect/sign-client and @web3modal/standalone

### DISCUSSION THREAD

**Discussion by ntoine223** [WalletConnect/walletconnect-monorepo#2245] [Developer Support]
Hi 👋

I am currently trying to use the **eth_sign** method with **WalletConnect** to sign a message from a wallet application, here Spot.

To do this I took the [js example project][1] available on this repos with the chosen configuration (html-standalone).

I modify the code to first try the **personnal_sign** method with the [data from the documentation][2] And here is the code: 

```javascript:

const signButton = document.getElementById("sign-button");

const namespaces = {
  eip155: {
    methods: ["eth_sign","personal_sign"],
    chains: ["eip155:1"],
    events: ["accountsChanged"],},};

export let session = undefined;

connectButton.addEventListener("click", async () => {
  try {
    if (signClient) {
      const { uri, approval } = await signClient.connect({
        requiredNamespaces: namespaces,
      });
      if (uri) {
        await web3Modal.openModal({ uri });
        session = await approval();  // save session
        signButton.disabled = false;  // Update sign button state
        signButton.innerText = "Sign"; // Update sign button text
        web3Modal.closeModal();}}} 
      catch (err) {
    console.error(err);}});

//  Sign message handler with personal_sign
signButton.addEventListener("click", async () => {
  try {
    if (signClient) {
      let params = ["0xdeadbeaf",session.namespaces.eip155.accounts[0].split(':')[2],]
      let method = "personal_sign"
      const result = await signClient.request({
        topic: session.topic,
        chainId: "eip155:1",
        request: {
          id: 1,
          jsonrpc: "2.0",
          method: method,
          params: params,
        },});
      if (result) {
        console.log(result);}}
  } catch (err) {
    console.error(err);}});
```

And this works correctly, I do have the modal opening in my wallet application and then I do get my message signature in reply.

However when I try to use the **eth_sign** method with the [data from the documentation][3] I don't have a working behavior. And here is the code: 

```javascript:
//  Sign message handler with eth_sign
signButton.addEventListener("click", async () => {
  try {
    if (signClient) {
      let params = [session.namespaces.eip155.accounts[0].split(':')[2],"0xdeadbeaf",]
      let method = "eth_sign"
      const result = await signClient.request({
        topic: session.topic,
        chainId: "eip155:1",
        request: {
          id: 1,
          jsonrpc: "2.0",
          method: method,
          params: params,
        },});
      if (result) {
        console.log(result);}}
  } catch (err) {
    console.error(err);}});
```

When the click is done, an error mode appears in my wallet application (Spot): 

> An error has occurred please try again later.

Mobile error : https://i.stack.imgur.com/MF87w.jpg

In the console of my browser I also have an error message which is very brief: 

> {context: 'client'} 'error'

Desktop error : https://i.stack.imgur.com/x7tzo.png

I tried to take several other wallet application compatible with **WalletConnect v2** but nothing makes it. I can't understand why the **eth_sign** request doesn't pass with the example data while **personal_sign** does.

A sentence in the documentation caught my attention, it is present in the Note of the [documentation][3] of **eth_sign** and of the other **eth_XX methods**. Unfortunately I don't understand the meaning of this sentence although I think it is an interesting lead: 

> Note the address to sign with must be unlocked.

If someone who has already used this tool would have a lead I would obviously be interested.

Thanks

  [1]: https://github.com/WalletConnect/web3modal-examples/tree/main/html-standalone
  [2]: https://docs.walletconnect.com/2.0/advanced/rpc-reference/ethereum-rpc#personal_sign
  [3]: https://docs.walletconnect.com/2.0/advanced/rpc-reference/ethereum-rpc#eth_sign

**Comment by finessevanes** [comment]
@0xAsimetriq could you take a look please.

**Comment by xzilja** [comment]
We don't manipulate these methods / requests in any way, so there could be few issues at play here:

1. Message in request is not formatted correctly
2. Spot doesn't support the method / payload structure (which is unlikely for popular method like this)
